File: leetcode/happynum.go
package main

import "math"

func isHappy(n int) bool {
	if n == 1 {
		return true
	}

	if n == 58 {
		return false
	}

	var s = 0
	for n != 0 {
		s += int(math.Pow(float64(n%10), 2))
		n = n / 10
	}

	return isHappy(s)
}

File: leetcode/happynum.py
class Solution(object):
    def isHappy(self, n):
        """
        :type n: int
        :rtype: bool
        """
        if n == 1:
            return True
        if n == 58:
            return False

        s = 0
        while (n):
            s += (n % 10)**2
            n = n / 10
        return self.isHappy(s)
